crates/assists/src/utils.rs

```diff
@@ -250,6 +266,25 @@ fn invert_special_case(expr: &ast::Expr) -> Option<ast::Expr> {
     }
 }
 
+fn bin_impls_ord(sema: &Semantics<RootDatabase>, bin: &ast::BinExpr) -> bool {
+    if let (Some(lhs), Some(rhs)) = (bin.lhs(), bin.rhs()) {
+        return impls_ord(sema, &lhs) && impls_ord(sema, &rhs);
```

Comment:
I think we want to check if lhs and rhs have the same type, then check that the type of them implements `Ord`, since `Ord` can only be implemented on types that implement `PartialOrd<Self>`.

Reply:
Otherwise we could incorrectly assume that `T` and `U` have total ordering with each other even though their types differ.

Reply:
I bet a solution is for `ìmpls_ord` to return the actual type autoderef-ed into and then I can compare the return values.

Reply:
I think we should just take the approach I mentioned, check that the types are equal and check if the autoderef-ed type implements Ord.

Given rust complains about `32isize == &32isize` for example I believe it doesn't do any fancy things with operators atm. Maybe @flodiebold knows more here.